### DISCUSSION TITLE
How to append a local image file to a page per blocks.children.append method?

### DISCUSSION THREAD

**Discussion by RickVincent** [jheddings/notional#54] [Q&A] [answered]
I am trying to upload an local image file (in my HDD) to an existing page, per below method.
notion.**blocks.children.append**(some_page, **blocks.Image**[_local_file_url_])

Example of the local_file_url looks like: -- Directory  = r"C:\Images"

But always got below failure. But if replace the local_file_url with a file URL from website (with https), it works.
_Exception has occurred: NotImplementedError
<class 'notional.blocks.Image'> does not support object composition_

Also tried replacing Image object with types.ExternalFile(local_file_url), still no luck.

per Notion's API doc: # https://developers.notion.com/reference/block#file-blocks, external file object can be block to be appended.
My understanding is so call "external file" can be file from local HDD as well. Am I right?

Any idea to append local image file to a page? or even as a property of the page in "Files & media" schema
![Error](https://user-images.githubusercontent.com/77380202/172118923-1bb7e2b6-1317-4ea1-ac77-868d3005fb23.png)
?

**Comment by jheddings** [comment, marked as answer]
Unfortunately, Notion will not let you use a locally hosted file as a URL.  The typical format would be `file:///path/to/file.jpg`.  If you try to add a block using a local path, you will receive and `Invalid image url` error response.

In the case where you want to reference an image from another website, you can build the `Image` block from a `FileObject`.  For example:

```python
file = types.ExternalFile[url]
img = blocks.Image(image=file)
notion.blocks.children.append(some_page, img)
```

I'll think about a better way for composing `Image` blocks directly from a URL.  Since you cannot upload content using the API, I've held off spending much time in this area.

**Comment by RickVincent** [comment]
Appreciate your answer and comment! And, your Notional API is awesome with really handy and convenient way to let developers can focus on app and logic, instead of building the payload for Notion API. Keep going...

**Reply by jheddings** [reply]
Thanks for the kind words!  I will definitely keep improving as time permits.
